CHANGELOG.md

```diff
@@ -3,6 +3,88 @@
+
+  As a first step for [#3066](https://github.com/mui-org/material-ui-x/issues/3066), these slots were renamed to clearly indicate that they are meant to replace a core component.
+
+  ```diff
+   <DataGrid
+    components={{
+  -    checkbox: MyCustomCheckbox,
+  +    BaseCheckbox: MyCustomCheckbox,
+    }}
+    componentsProps={{
+  -    checkbox: {},
+  +    baseCheckbox: {},
+    }}  
+  />;
+  ```
+
+**Note**: these changes apply to both the `DataGrid` and `DataGridPro` components.
+
+#### Changes
+
+- [DataGrid] Block multi-rows updates in `apiRef.current.updateRows` on the Community plan (#3095) @flaviendelangle
+- [DataGrid] Fix filter not working after deleting the value (#3018) @m4theushw
+- [DataGrid] Fix performance regression when selecting 100k rows (#3077) @m4theushw
+- [DataGrid] Fix `apiRef.current.updateRows` to not share rows from other instances (#3127) @m4theushw
+- [DataGrid] Fix flex space allocation to not cause a horizontal scroll when there is enough space (#3099) @flaviendelangle
+- [DataGrid] Improve the filter panel behaviors (#3080) @flaviendelangle
+- [DataGrid] Fix keyboard navigation between column headers and rows when not on the first page (#3086) @flaviendelangle
+- [DataGrid] Fix keyboard navigation between rows when not on the first page (#3074) @flaviendelangle
+- [DataGrid] Prevents bubbling in menu header (#3000) @alexfauquette
+- [DataGrid] Remove unused rendering state and selectors (#3133) @flaviendelangle
+- [DataGrid] Rename `Checkbox` component and props slots to `BaseCheckbox` (#3142) @DanailH
+
+### Core
+
+- [core] Automatically close issues that are incomplete and inactive (#3029) @oliviertassinari
+- [core] Improve the typing of `LicenseStatus` (#3141) @Himself65
+- [core] Make `useGridColumnsPreProcessing` generic (#3092) @m4theushw
+- [core] Move column headers virtualization to hook (#3078) @m4theushw
+- [core] Move virtualization logic to hook (#3079) @m4theushw
+- [core] Rename directories to match new packages new names (#3088) @flaviendelangle
+- [core] Replace `createClientRender` with new `createRenderer` API (#3125) @flaviendelangle
+- [core] Store the event manager in a key of `GridApi` instead of making the whole `GridApi` extend it (#3069) @flaviendelangle
+- [core] Update monorepo (#3139) @m4theushw
+- [core] Use `unstable_` prefix instead of `unsafe_` for private APIs (#3090) @flaviendelangle
+- [core] Use official MUI repo as monorepo (#3084) @m4theushw
+
+### Docs
+
+- [docs] Fix broken example in the component slot example (#3123) @Himself65
+- [docs] Fix inline previews (#3081) @DanailH
+- [docs] Fix the client-side validation link clarity (#3100) @oliviertassinari
+- [docs] Improve `rowCount` CSS class description (#3072) @ZeeshanTamboli
+- [docs] Use core repo constants for doc internationalization (#3143) @flaviendelangle
+- [core] Adapt changelog script to GitHub DOM modification (#3087) @alexfauquette
```

Comment:
This one should not be here if it's "core".  https://github.com/mui-org/material-ui-x/pull/3160#discussion_r747500087